src/components/pagination/OcPagination.vue

```diff
@@ -0,0 +1,337 @@
+          :class="pageClass(page)"
+          v-bind="pageProps(page)"
+          v-text="page"
+        />
+      </li>
+      <li v-if="isNextPageAvailable" class="oc-pagination-list-item">
+        <router-link
+          class="oc-pagination-list-item-next"
+          :aria-label="$gettext('Go to the next page')"
+          :to="nextPageLink"
+        >
+          <oc-icon name="chevron_right" />
+        </router-link>
+      </li>
+      <li v-if="isGoToLinkVisible" class="oc-pagination-list-item-goto">
+        <oc-text-input
+          v-model="goToTarget"
+          :label="$gettext('Go to')"
+          class="oc-pagination-list-item-goto-input"
+        />
+        <component
+          :is="goToLinkComponent"
+          v-bind="goToLinkProps"
+          class="oc-pagination-list-item-goto-link"
+        >
+          <oc-icon name="chevron_right" />
+        </component>
+      </li>
+    </ul>
+  </nav>
+</template>
+
+<script>
+import OcIcon from "../OcIcon.vue"
+
+/**
+ * A list of links used for switching to different pages
+ */
+export default {
+  name: "OcPagination",
+
+  components: { OcIcon },
+
+  props: {
+    /**
+     * Number of pages to be displayed
+     */
+    pages: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Currently active page
+     */
+    currentPage: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Number of pages to be displayed. Needs to be an even number.
+     * Pages will be equaly split into two parts and remaining pages will be displayed as ellipsis
+     */
+    maxDisplayed: {
+      type: Number,
+      required: false,
+      default: null,
+      validator: value => {
+        if (value % 2 === 0) {
+          return true
+        }
+
+        // Since Vue doens't support custom validator error message, log the error manually
+        console.error("maxDisplayed needs to be an even number")
+      },
+    },
+    /**
+     * Current route which is used to render pages
+     */
+    currentRoute: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data: () => ({
+    goToTarget: "",
+  }),
+
+  computed: {
+    displayedPages() {
+      if (this.maxDisplayed && this.maxDisplayed < this.pages) {
+        let pages = []
+
+        for (let i = 0; i < this.pages; i++) {
+          pages.push(i + 1)
+        }
+
+        const pagesRight = pages.slice(this.currentPage, this.currentPage + this.maxDisplayed / 2)
+        const pagesLeft = pages.slice(
+          this.currentPage - 1 - this.maxDisplayed / 2,
+          this.currentPage - 1
+        )
+
+        pages = [...pagesLeft, this.currentPage, ...pagesRight]
+
+        pages[0] > 2 ? pages.unshift(1, "...") : pages.unshift(1)
+        pages[pages.length - 1] < this.pages - 1
+          ? pages.push("...", this.pages)
+          : pages.push(this.pages)
+
+        return pages
+      }
+
+      return this.pages
+    },
+
+    isPrevPageAvailable() {
+      return this.currentPage !== 1
```

Comment:
```suggestion
      return this.currentPage > 1
```
to make it more resilient (e.g. when page number comes from the url)